polyfill/lib/ecmascript.mjs

```diff
@@ -2784,9 +2783,13 @@ export function RejectToRange(value, min, max) {
   if (value < min || value > max) throw new RangeErrorCtor(`value out of range: ${min} <= ${value} <= ${max}`);
 }
 
-export function RejectISODate(year, month, day) {
-  RejectToRange(month, 1, 12);
-  RejectToRange(day, 1, ISODaysInMonth(year, month));
+export function RejectISODate(year, month, day, overflowMonths, overflowDays) {
+  if (overflowMonths === 'reject') {
+    RejectToRange(month, 1, 12);
+  }
+  if (overflowDays === 'reject') {
+    RejectToRange(day, 1, ISODaysInMonth(year, month));
+  }
```

Comment:
Likewise, RejectISODate can just stay as is.

Reply:
It doesn't work with the implementation of RejectISODate being as it is, because there's no way for it to distinguish between the case where it should (effectively) constrain the day despite `overflow` being `reject`, and the case where it should reject the day, without taking an overflow argument.